### ISSUE TITLE
[bug] Incorrect message serialization

### ISSUE THREAD

**Issue by drygdryg** [PMunch/protobuf-nim#23]
I tried serializing and dump it into a file the following message:
```protobuf
syntax = "proto3";

message Example2 {
    string field1 = 1;
    int32 field2 = 2;
}

message Example {
    message ExampleNested {
        int32 field1 = 1;
        repeated Example2 example2 = 2;
        int32 field3 = 3;
    }
    ExampleNested exampleNested = 1;
    int32 field2 = 2;
}
```
Here's the source:
```nim
import streams
import protobuf

const protoSpec = """
syntax = "proto3";

message Example2 {
    string field1 = 1;
    int32 field2 = 2;
}

message Example {
    message ExampleNested {
        int32 field1 = 1;
        repeated Example2 example2 = 2;
        int32 field3 = 3;
    }
    ExampleNested exampleNested = 1;
    int32 field2 = 2;
}
"""
parseProto(protoSpec)

var msg = new Example
msg.exampleNested = initExample_ExampleNested(field1 = 5, field3 = 6)
let example2 = initExample2(field1 = "Test", field2 = 123)
msg.exampleNested.example2 = @[example2]
msg.field2 = 5

var strm = newFileStream("data_bug.bin", fmWrite)
strm.write(msg)
```
After which I tried to decode it with protoc and got the error:
```
$ protoc --decode_raw < data_bug.bin
Failed to parse input.
```
To be sure, I implemented the same in Python and found no error.
I attach all the files needed to verify: https://gist.github.com/drygdryg/19319a61ca9a7c83f251b63946ed86c2

**Comment by drygdryg** [comment]
I also compared the output with vbindiff and noticed that one byte is different.
Correct serialization:
![image](https://user-images.githubusercontent.com/43933400/103396409-2f65eb80-4b44-11eb-9454-a391803a0b4d.png)
Incorrect:
![image](https://user-images.githubusercontent.com/43933400/103396421-3e4c9e00-4b44-11eb-942e-72a12abf9e23.png)

**Comment by PMunch** [comment]
After boiling this down a bit further I managed to figure out what the error was. It simply didn't account for the written size of sub-objects. I implemented a "fix" but while writing this I realise that it doesn't work properly as it doesn't take into account the variable length of integers. Shouldn't be too hard to fix, but I haven't got time for it right now..

**Comment by PMunch** [comment]
This is now properly fixed in v0.5.1

**Comment by drygdryg** [comment]
Thank you for your quick response. Indeed, this bug is fixed, but I found a more complicated case with repeated fields where the bug still persists.
```nim
import streams, protobuf

const protoSpec = """
syntax = "proto3";

message WifiDeviceMsg {
    string bssid_string = 1;
    uint64 bssid_decimal = 8;
}

message RequestMsg {
    message PlatformProfileMsg {
        string version = 1;
        string platform = 2;
        string locale = 5;
    }
    PlatformProfileMsg platformProfile = 1;

    message RequestElementsMsg {
        message WifiProfileMsg {
            int64 timestamp = 1;
            repeated WifiDeviceMsg wifiDevice = 2;
            int32 unknown3 = 3;
        }
        WifiProfileMsg wifiProfile = 2;
        int32 unknownMessage10 = 10;
    }
    repeated RequestElementsMsg requestElements = 4;
}
"""
parseProto(protoSpec)

let message = new RequestMsg
message.platformProfile = initRequestMsg_PlatformProfileMsg(
  version = "2021",
  platform = "android/LEAGOO/full_wf562g_leagoo/wf562g_leagoo:6.0/MRA58K/1511161770:user/release-keys",
  locale = "en_US")
var element = initRequestMsg_RequestElementsMsg(unknownMessage10 = 0'i32)
element.wifiProfile = initRequestMsg_RequestElementsMsg_WifiProfileMsg(
    timestamp = 162723'i64, wifiDevice = @[], unknown3 = 2'i32)
element.wifiProfile.wifiDevice.add(initWifiDeviceMsg(
    bssid_string = "",
    bssid_decimal = 212195084994643'u64))
element.wifiProfile.wifiDevice.add(initWifiDeviceMsg(
  bssid_string = "",
  bssid_decimal = 18838586676582'u64))
message.requestElements = @[element]
var stream = newFileStream("data.bin", fmWrite)
stream.write(message)
```
Tested it with:
```
$ protoc --decode_raw < data.bin
Failed to parse input.
```
